### ISSUE TITLE
Existing test suite is failing

### ISSUE THREAD

**Issue by erikrorstrom** [HubSpot/hubspot-php#140]
Existing tests are failing on master, we should fix that so new PR's can pass travis CI

**Comment by ryanwinchester** [comment]
Yeah, we should get rid of all the tests hitting hubspot API and just mock them or something. They stopped working a long time ago.

**Comment by erikrorstrom** [comment]
@ryanwinchester Atleast deleting the phpspec tests that seem outdated?

The failing tests in phpunit is actualy failing, those can be fixed

**Comment by Pryrios** [comment]
I don't think that a mocking will help because Hubspot is not very good at documenting their API. The mock will probably not reflect correctly its functionality so the tests will not be useful and the library may still fail on a production environment after all.

I've tried to update the tests by using my developer account. Using dev accounts would mean that to run the tests everyone has to get their own developer account, create an API Key and configure it on the tests.

Furthermore, the evolution of the Hubspot API has implications on the current tests. For example, testing the blog posts now require that:

- The blog posts need to have an author id.
- The blog posts need to have a metatag description.
- The blog posts need to have an url (slug).
- You need to know the Blog id on your development account.
- Other elements related to blog posts deletion, blog posts versioning and so on.

I haven't been able to fully fix those tests as for example deleting a post returns me a "405 Not Allowed" but I am following documentation to the letter. So I don't really have a clue about what's really happening. Many of those things aren't even documented, you just need to know them from normal Hubspot usage, such as that the author or metatag description are a required fields on blog posts.

And this is just one of the APIs.

At this point I don't think the real API and the demo environments are useful for CI automated testing.

In my opinion, I think that we should keep the tests related to Hubspot outside of CI, but I think think it would still be useful to fix them and leave them for people to use on their own environments and document that on the README.

I'll try to do so on my fork on the coming days.

**Comment by zman81988** [comment]
Hey @Pryrios I'm Zack, from the HubSpot developer relations team. I want to make sure we have our documentation as up to date as possible.  I just tried the delete blog post request and got the expected 204 no content response following this [doc](https://developers.hubspot.com/docs/methods/blogv2/delete_blog_posts_blog_post_id). Could you point me to the docs you followed that lead to the 405?  I want to make sure it's either taken down or updated as appropriate.  

In general, if you see anything off on our documentation, feel free to `@mention` me in this repo and I'll get our docs sorted.

**Comment by Pryrios** [comment]
@zman81988 I didn't know you were hanging around here, that's really wonderful! Double-wonderful as you are committed to cleaning up and updating the docs! :)

I think I've had the issue with restoring a previously deleted blog post https://developers.hubspot.com/docs/methods/blogv2/post_blog_posts_blog_post_id_restore_deleted

Let me take another look at it to see if I can give you proper data (parameters, responses and such) of the specific failing test. If there is some kind of support from Hubspot it will make it easier to update this repo and bring it up to speed.

I'll get back to you in some minutes.

**Comment by Pryrios** [comment]
@zman81988 effectively, it is when trying to restore a deleted blog post. The Test does the following:

1. Create a Blog Post - OK
2. Publish it - OK
3. Delete it - OK
4. Restore the deleted version <-- This is what fails

The Rest call is as follows:

POST https://api.hubapi.com/content/api/v2/blog-posts/10258255669/restore-deleted?hapikey=<my demo API key>

**Comment by zman81988** [comment]
@Pryrios My team is relatively new to HubSpot so I haven't been hanging out here long so there might be some older issues that needed a documentation fix that I'm not aware of if you happen to know of anything definitely let me know.   

Thanks for putting that list together (and thank you for not putting your hapikey into the comment, you'd be surprised how many times we've had to tell people not to do that :) ).  I've updated the documentation [here](https://developers.hubspot.com/docs/methods/blogv2/post_blog_posts_blog_post_id_restore_deleted?hsCacheBuster=2241) to reflect that the correct HTTP method is PUT for that endpoint, rather than the previously documented POST.  If you update your tests to use PUT instead of POST that should resolve the 405 error you were seeing.
